forte/predictor.py

```diff
@@ -0,0 +1,185 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import Dict, List, Callable
+import itertools
+from torch.nn import Module
+from forte.data.types import DATA_INPUT
+from forte.data.data_pack import DataPack
+from forte.data.ontology import Annotation
+from forte.processors.base.base_processor import BaseProcessor
+from forte.process_manager import ProcessJobStatus
+
+
+class Batcher:
+    '''This class will create a pool of pack, instance and feature. When
+    the pool is filled with a batch size of elements. It will generate them
+    in a batch. The batched data will be pass to the prediction function in the
+    Predictor. Note that the extract, convert process is done during
+    this batching process.
+    '''
+    def __init__(self, batch_size: int, feature_resource: Dict,
+                cross_pack=True):
+        # pylint: disable=line-too-long
+        '''Feature_resource is prodcued from train pipeline.
+        An example looks like
+        {
+        'scope': ft.onto.base_ontology.Sentence,
+        'schemes': {
+            'text_tag': {
+                'extractor': <forte.data.extractor.extractor.TextExtractor>,
```

Comment:
I have a trick that I learned. We can simply use the name "TextExtractor", but automatically search it in "forte.data.extractor.extractor" package (we can further define the packages to search). This will make the config looks even shorter.

Reply:
This is very interesting. I guess you mean specifying a string refering to the Extractor class in the user request, right?

My question how much will this simply the request compared with current design. Here is an [example](https://github.com/asyml/forte/pull/318/files#diff-8e0947db70ddeea5344d86bb30629889ce5851b535e96e7fa5d36f681fc42a1aR150) request that user will write. As you can see, currently user will import the Extractor module and directly send the class type into request. It seems like not too hard to write?

Reply:
I see, it looks like the `class_name` is already simplified here.